repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java

```diff
@@ -444,9 +444,11 @@ public void deleteUser( Session session, final IPentahoUser user ) throws NotFou
         while ( currentGroups.hasNext() ) {
           currentGroups.next().removeMember( jackrabbitUser );
         }
+        getUserCache().remove( jackrabbitUser.getID() );
         purgeUserFromCache( user.getUsername() );
```

Comment:
There was already a call to remove the user from the cache. Now we're removing it twice?

Reply:
The purgeUserFromCache() method name is a misnomer, that method only removes the user from the userDetailsCahche and not the userCache. I thought of modifying the method to clear both but you need the jackrabit user ID to remove the user from the userCache and some of the places where this method is called from don't have the jackrabit user available. Also, I didn't know if the places where they called this method onle needed to remove the user from both caches or would I break something, that is why I added the userCache remove outside the misnamed method.